### PR TITLE
Djid functionality added

### PR DESCRIPTION
**Pull request by zefciu**
* The new bob.djid module is added
* Existing modules are modified to use require.js
* Tests of existing js modules are done via jasmine
* First steps of a tutorial are created

### CODE REVIEW

File: bob/djid/__init__.py [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+    queryset = djid._meta.Model.objects.all()
+    queryset.query = query
+    return djid.get_export_data(djid, queryset, content_type)
+
+
+
+
+#     
+#     def get_response(self, request, djid, *args, **kwargs):
+#         return HttpResponse(data, content_type='text/csv')
```
Comment (ar4s) [inline bob/djid/__init__.py]: Don't be afraid remove code :)

File: bob/djid/static/js/djid.js [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+             this.remove_progress();
+             bootbox.alert('Wygenerowanie wyniku nie powiodło się!');
+             return;
+        }
+        // if (result.progress) {
+        //     this.set_progress(result.progress);
+        // }
+        // if (result.eta) {
+        //     this.set_eta(result.eta);
+        // }
```
Comment (ar4s) [inline bob/djid/static/js/djid.js]: DITO

File: bob/djid/static/js/djid.js [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+    Djid.prototype.handle_update = function (result) {
+        var that, data;
+        that = this;
+        if (result.failed) {
+             clearInterval(this.long_interval_handle);
+             clearInterval(this.short_interval_handle);
+             this.set_eta(null);
+             this.remove_progress();
+             bootbox.alert('Wygenerowanie wyniku nie powiodło się!');
```
Comment (ar4s) [inline bob/djid/static/js/djid.js]: polish string

File: bob/djid/static/js/djid.js (hunk 2 above)
Comment (ar4s) [inline bob/djid/static/js/djid.js]: This commented block is necessary?

File: djid-tasks [hunk 4]
```diff
@@ -0,0 +1,5 @@
+Zadania do djida:
```
Comment (xliiv) [inline djid-tasks]: This fiel is probably mistake. ;)

File: bob/djid/templates/djid.html [hunk 5]
```diff
@@ -0,0 +1,16 @@
+<table id="{{meta.djid_id}}" class="djid"><tr><td></td></tr></table>
+<div id="{{meta.djid_id}}-pager"></div>
```
Comment (ar4s) [inline bob/djid/templates/djid.html]: pls insert space after ``{{`` and before ``}}``

File: bob/test_djid/tests/test_class.py [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+            class Meta:
+                djid_id = 'persons'
+                Model = Person
+                query_set = Person.objects.all()
+                columns = [
+                    'first_name',
+                    'last_name',
+                    'registered',
+                ]
+            registered=DateTimeColumn(label='Zarejestrowany')
```
Comment (xliiv) [inline bob/test_djid/tests/test_class.py]: translate it, please

File: bob/test_djid/tests/__init__.py [hunk 7]
```diff
@@ -0,0 +1,3 @@
+from bob.test_djid.tests.test_class import *
```
Comment (xliiv) [inline bob/test_djid/tests/__init__.py]: I think we don't use '*' imports.

File: bob/djid/column/__init__.py [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+    def get_model(self):
+        """Return a dict for jqgrid colModel"""
+        result = {"name": self.name}
+        if self.as_link:
+            result['formatter'] = 'djid_link'
+        result['search'] = self.filtered
+        return result
+
+    def format_label(self, model):
+        """Return the text to be displayed in the cell."""
```
Comment (xliiv) [inline bob/djid/column/__init__.py]: ?

File: bob/djid/__init__.py [hunk 9]
```diff
@@ -0,0 +1,306 @@
...
+        # No. Removing the dictionary below is not a good idea, as it will
+        # allow user to call any method of djid
+        method = getattr(djid, {
+            'get_ajax': 'get_ajax_data',
+            'start_report': 'start_report',
+            'update_status': 'update_status',
+            'get_report': 'get_report',
+        }[action])
+        
+        decorated_method = types.MethodType(
```
Comment (xliiv) [inline bob/djid/__init__.py]: This needs explanation in comment.

File: bob/test_djid/tests/__init__.py (hunk 7 above)
Comment (zefciu) [inline bob/test_djid/tests/__init__.py]: It's __init__.py and we need it for tests

File: bob/djid/column/__init__.py (hunk 8 above)
Comment (zefciu) [inline bob/djid/column/__init__.py]: !

**Comment by xliiv** [comment]
:ok:

**Comment by ar4s** [comment]
:ok:

File: bob/djid/column/__init__.py [hunk 10]
```diff
@@ -0,0 +1,252 @@
...
+from bob.djid.util import PEP3115
+
+
+if not PEP3115:
+    _counter = 0
+
+
+class Column(object):
+    """A column object.
+        :param as_link: True or False to force creating links to related object
```
Comment (deejay1) [inline bob/djid/column/__init__.py]: what doc string style is it? especially that a different style is used below. Nice guidelines are used by https://github.com/openstack-dev/hacking/blob/master/hacking/checks/docstrings.py

File: docs/djid_tutorial.rst [hunk 11]
```diff
@@ -0,0 +1,221 @@
+Djid tutorial
+=======================
+
+Introduction
+--------------------
+
+Djid is a framework for tabular presentation that facilitates
+setting up a grid basing on django models. It utilises jqGrid_ on the browser
```
Comment (deejay1) [inline docs/djid_tutorial.rst]: *based on*

File: docs/djid_tutorial.rst [hunk 12]
```diff
@@ -0,0 +1,221 @@
...
+side. Its features are:
+
+    * AJAX data - all the data is loaded via AJAX_ request in json_ format.
+    * pagination - your data will be paginated and automatically loaded when
+      the user scrolls your grid
+    * filtering - the user can filter data with an automatically generated form
+    * sorting - it is possible to order the data by several columns
+    * extendability - you can create your own columns that encapsulate
```
Comment (deejay1) [inline docs/djid_tutorial.rst]: didn't you mean extensibility?

File: docs/djid_tutorial.rst [hunk 13]
```diff
@@ -0,0 +1,221 @@
...
+                        'jquery',
+                        'jquery-ui'
+                    ]
+                },
+            }
+        });
+    </script>
+
+Fortunately you only need to do this once for your whole application.
+You only need to specify ``paths`` only if your javascript files are not to be
```
Comment (deejay1) [inline docs/djid_tutorial.rst]: one *only* should be enough for everyone

File: docs/api.rst [hunk 14]
```diff
@@ -0,0 +1,12 @@
+Api reference
+===============================
```
Comment (deejay1) [inline docs/api.rst]: Try to keep the section headers (==== etc) only as long as the above string, it looks nicer IMNSHO

File: docs/djid_tutorial.rst (hunk 12 above)
Comment (zefciu) [inline docs/djid_tutorial.rst]: http://www.oxforddictionaries.com/definition/english/extend

File: docs/djid_tutorial.rst (hunk 11 above)
Comment (zefciu) [inline docs/djid_tutorial.rst]: "basing" is a present particible not a gerund

File: bob/djid/column/__init__.py (hunk 10 above)
Comment (zefciu) [inline bob/djid/column/__init__.py]: It's a sphinx docstring style

Comment (deejay1) [inline bob/djid/column/__init__.py]: With that weird indentation?

**Pull request merged**